### ISSUE TITLE
Website verification link is case sensitive

### ISSUE THREAD

**Issue by edent** [mastodon/mastodon#18308]
### Steps to reproduce the problem

1. I added `<a rel="me" href="https://mastodon.social/@edent">Mastodon</a>` to my website.
2. The linkback verification didn't work, because my username is case-sensitive and should be **E**dent.
3. Changing the link to `<a rel="me" href="https://mastodon.social/@Edent">Mastodon</a>` made it work.

I didn't notice the need for a case sensitive name, because the text input field cuts off the full link
![Screenshot 2022-05-04 at 17-52-22 Edit profile - Mastodon](https://user-images.githubusercontent.com/837136/166740538-80e0c655-8aba-4c3b-af80-e9a8cbaa5107.png)

### Expected behaviour

Given that usernames generally don't behave in a case sensitive manner, I think the link verification service should be more liberal iin what it accepts.

### Actual behaviour

The link checker is case sensitive.

### Specifications

Whatever is running mastodon.social ☺

I think the code is at 

https://github.com/mastodon/mastodon/blob/db01df2ddabdc39afecdbf75ccd0b4ecc292240c/app/services/verify_link_service.rb#L31

My Ruby isn't great, but I think you can use `casecmp` here - https://ruby-doc.org/core-3.1.2/String.html#method-i-casecmp-3F

Alternatives:

* Change the `<input>` to a `<textarea>` - although that would mean changing the JS for copying the link. 
* Making the hint explicit that the user name is case sensitive.

**Comment by ClearlyClaire** [comment]
Since Mastodon serves the `@link_back` itself and Mastodon usernames are explicitly case-insensitive since a long time ago, this change should be safe. Mind opening a PR?

**Comment by emceeaich** [comment]
That looks fine to me, @edent. Testing in irb.

```
irb(main):007:1* links = [
irb(main):008:0> {"href" =>  "https://foo.bar/@emma"}, {"href" => "https://foo.bar/@Emma"}, {"href" => "https://f
oo.bar/@Bob"}]
=> [{"href"=>"https://foo.bar/@emma"}, {"href"=>"https://foo.bar/@Emma"}, {"href"=>"https://foo.bar/@Bob"}]
irb(main):009:0> links.any? { |link| link["href"].casecmp?("https://foo.bar/@emma") }
=> true
```

I'd make a PR.

**Comment by ClearlyClaire** [comment]
Fixed by #18320
